### PR TITLE
chore: use retrieveRequestParams where possible

### PR DESCRIPTION
**Pull request by kangmingtay**
## What kind of change does this PR introduce?
* Move the reading the requestBody into the captcha middleware, which is in the api package so it can use `retrieveRequestParams` 
* Use `retrieveRequestParams` in the admin delete route

## What is the current behavior?

Please link any relevant issues here.

## What is the new behavior?

Feel free to include screenshots if it includes visual changes.

## Additional context

Add any other context or screenshots.

### CODE REVIEW

File: internal/api/middleware.go [hunk 1]
```diff
@@ -116,12 +117,13 @@ func (a *API) verifyCaptcha(w http.ResponseWriter, req *http.Request) (context.C
 		return ctx, nil
 	}
 
-	verificationResult, err := security.VerifyRequest(req, strings.TrimSpace(config.Security.Captcha.Secret), config.Security.Captcha.Provider)
-	if err != nil {
-		if strings.Contains(err.Error(), "request body was not JSON") {
-			return nil, badRequestError(ErrorCodeValidationFailed, "Request body for CAPTCHA verification was not a valid JSON object")
```
Comment (kangmingtay) [inline internal/api/middleware.go:122]: we should use `ErrorCodeBadJSON` which is consistent with the error returned in `retrieveRequestParams`

File: internal/api/admin.go [hunk 2]
```diff
@@ -512,21 +511,18 @@ func (a *API) adminUserDelete(w http.ResponseWriter, r *http.Request) error {
...
-	body, err := getBodyBytes(r)
-	if err != nil {
-		return internalServerError("Could not read body").WithInternalError(err)
-	}
-	if len(body) > 0 {
-		if err := json.Unmarshal(body, params); err != nil {
-			return badRequestError(ErrorCodeBadJSON, "Could not read params: %v", err)
+	if err := retrieveRequestParams(r, params); err != nil {
+		if err.(*HTTPError).ErrorCode == ErrorCodeBadJSON {
+			// request body is empty so the default behavior should be to hard delete the user
```
Comment (J0) [inline internal/api/admin.go:518]: /nit Not sure if [Unmarshal returning an error](https://github.com/supabase/auth/pull/1818/files#diff-3e8b73f5901bb9b41436eb294ebc11f27d7f3f93d07c779b0be7a771af8a3bdfR99)  is always equivalent to body is empty

Can consider removing the comment

**Review by J0 — APPROVED** [review]
Nice, great work

File: internal/api/admin.go (hunk 2 above)
Comment (kangmingtay) [reply internal/api/admin.go:518]: hmm, good point, actually i think we shouldn't ignore the other errors - will add a new error code to handle this

File: internal/api/admin.go [hunk 3]
```diff
@@ -512,21 +512,17 @@ func (a *API) adminUserDelete(w http.ResponseWriter, r *http.Request) error {
...
+	// ShouldSoftDelete defaults to false
 	params := &adminUserDeleteParams{}
-	body, err := getBodyBytes(r)
-	if err != nil {
-		return internalServerError("Could not read body").WithInternalError(err)
-	}
-	if len(body) > 0 {
-		if err := json.Unmarshal(body, params); err != nil {
-			return badRequestError(ErrorCodeBadJSON, "Could not read params: %v", err)
+	if body, _ := utilities.GetBodyBytes(r); len(body) != 0 {
```
Comment (kangmingtay) [inline internal/api/admin.go:517]: we can ignore the error returned here and let `retrieveRequestParams` surface it - this is just to check quickly if the body is empty

**Pull request merged**